web-wallet/src/lib/containers/MigrateContract/MigrateContract.svelte

```diff
@@ -0,0 +1,468 @@
+    Icon,
+    Textbox,
+    Wizard,
+    WizardStep,
+  } from "$lib/dusk/components";
+  import {
+    mdiArrowRight,
+    mdiCheckDecagramOutline,
+    mdiInformationOutline,
+    mdiTimerSand,
+    mdiWalletOutline,
+  } from "@mdi/js";
+  import { createCurrencyFormatter } from "$lib/dusk/currency";
+  import { logo } from "$lib/dusk/icons";
+  import { calculateAdaptiveCharCount, middleEllipsis } from "$lib/dusk/string";
+  import { onMount } from "svelte";
+  import { settingsStore } from "$lib/stores";
+
+  const ERC_TOKEN = "ERC-20";
+  const BEP_TOKEN = "BEP-20";
+
+  const options = [
+    { disabled: false, label: ERC_TOKEN, value: ERC_TOKEN },
+    { disabled: false, label: BEP_TOKEN, value: BEP_TOKEN },
+  ];
+
+  /** @type {String} */
+  let selected = ERC_TOKEN;
+
+  /** @type {Boolean} */
+  const migrationInProgress = false;
+
+  /** @type {Boolean} */
+  let isWalletConnected = false;
+
+  /** @type {String} */
+  let connectedWalletAddress =
+    "6rmam3FisEHih84oJMEsmi2GGVetDPmcjw7H4heeVbm79DwZUPvPJTBeFriLx85Cy2Q2cGisz32BeSYX99kNXjD";
+
+  /** @type {Number} */
+  let connectedWalletBalance = 10;
+
+  /** @type {Number|String} */
+  let amount = "";
+
+  /** @type {HTMLInputElement | null} */
+  let amountInput;
+
+  /** @type {Boolean} */
+  let isMigrationInitialized = false;
+
+  /** @type {Number} */
+  const gasFee = 1;
+
+  /** @type {Boolean} */
+  let isMigrationBeingApproved = false;
+
+  /** @type {String}*/
+  const estimatedTime = "45 min";
+
+  /** @type {number} */
+  let screenWidth = window.innerWidth;
+
+  const { darkMode } = $settingsStore;
+
+  const minAmount = 0.000000001;
+
+  $: maxSpendable = connectedWalletBalance - gasFee;
+  $: isAmountValid =
+    typeof amount === "number"
+      ? amount >= minAmount && amount <= maxSpendable
+      : false;
+  $: ({ language } = $settingsStore);
+  $: duskFormatter = createCurrencyFormatter(language, "DUSK", 9);
+  $: if (selected) {
+    resetMigration();
+  }
+
+  onMount(() => {
+    amountInput = document.querySelector(".migrate__input-field");
+
+    const resizeObserver = new ResizeObserver((entries) => {
+      const entry = entries[0];
+
+      screenWidth = entry.contentRect.width;
+    });
+
+    resizeObserver.observe(document.body);
+
+    return () => resizeObserver.disconnect();
+  });
+
+  /**
+   * @param {Number} ms
+   */
+  function delay(ms) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  function resetMigration() {
+    if (isWalletConnected) {
+      isWalletConnected = false;
+      connectedWalletAddress = "";
+      connectedWalletBalance = 10;
+      amount = "";
+      isMigrationInitialized = false;
+      isMigrationBeingApproved = false;
+    }
+  }
+</script>
+
+<article class="migrate">
+  <header class="migrate__header">
+    <h3 class="h4">Migrate</h3>
+    <div class="migrate__header-icons">
+      <div>
+        <AppImage
+          src={darkMode ? "/binance_dusk_light.svg" : "/binance_dusk.svg"}
```

Comment:
Are these checks correct? You're using the "light" version of the images in the dark mode.
Maybe is the svg file that has a confusing name instead?

Reply:
I was being consistent, taking as example the way the dusk logo SVG are named.